src/cli.rs

```diff
@@ -619,50 +687,183 @@ fn construct_url(
+            }
+        }
+        roff.control("TP", []);
+        roff.text(header);
+        roff.text(body);
     }
+
+    if let Some(opt) = items.iter().find(|opt| opt.get_id() == "raw-method-or-url") {
+        roff.control("TP", []);
+        roff.text(vec![
+            roman("["),
+            italic("METHOD"),
+            roman("]"),
+            italic(" URL"),
+        ]);
+        if let Some(help) = opt.get_long_help().or_else(|| opt.get_help()) {
+            roff.text(vec![roman(help)]);
+        }
+    }
+
+    if let Some(opt) = items.iter().find(|opt| opt.get_id() == "raw-rest-args") {
+        roff.control("TP", []);
+        roff.text(vec![roman("["), italic("REQUEST_ITEM"), roman(" ...]")]);
+        if let Some(help) = opt.get_long_help().or_else(|| opt.get_help()) {
+            // replace the indents in request_item help with proper roff controls
+            // For example:
+            //
+            // ```
+            // normal help normal help
+            // normal help normal help
+            //
+            //   request-item-1
+            //     help help
+            //
+            //   request-item-2
+            //     help help
+            //
+            // normal help normal help
+            // ```
+            //
+            // Should look like this with roff controls
+            //
+            // ```
+            // normal help normal help
+            // normal help normal help
+            // .RS 12
+            // .TP
+            // request-item-1
+            // help help
+            // .TP
+            // request-item-2
+            // help help
+            // .RE
+            //
+            // .RS
+            // normal help normal help
+            // .RE
+            // ```
+            let lines: Vec<&str> = help.lines().collect();
+            let mut rs = false;
+            for i in 0..lines.len() {
+                if lines[i].is_empty() {
+                    let prev = lines[i - 1].chars().take_while(|&x| x == ' ').count();
+                    let next = lines[i + 1].chars().take_while(|&x| x == ' ').count();
+                    if prev != next && next > 0 {
+                        if !rs {
+                            roff.control("RS", ["12"]);
+                            rs = true;
+                        }
+                        roff.control("TP", []);
+                    } else if prev != next && next == 0 {
+                        roff.control("RE", []);
+                        roff.text(vec![roman("")]);
+                        roff.control("RS", []);
+                    } else {
+                        roff.text(vec![roman(lines[i])]);
+                    }
+                } else {
+                    roff.text(vec![roman(lines[i].trim())]);
+                }
+            }
+            roff.control("RE", []);
+        }
+    }
+
+    let mut manpage = fs::read_to_string(format!("{}/man-template.roff", rest_args[0])).unwrap();
```

Comment:
Maybe we could `include_str!` it instead?

Reply:
I think `include_str!` will read the file at compile, but the template path is determined at runtime.

Reply:
Do we need to determine it at runtime, though? Is there a case where you'd want to use a different version of the template?

Reply:
Hmm, while there is no reason to determine the path at runtime, I think it would be convenient to read `man-template.roff` at runtime to avoid recompiling xh whenever the template is edited.